### PR TITLE
WEBHOOK SUPPORT

### PR DESCRIPTION
**Pull request by andreasph**

### CODE REVIEW

File: example/webhook.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	if err != nil {
+		logger.Panicln(err)
+	}
+
+	for _, webhookEntity := range webhookEntities.WebhookEntities {
+		dumpWebhookEntity(webhookEntity, logger)
+	}
+
+	callbackUrl := "https://sellerapi.sellercenter.net/"
```
Comment (iru-b) [inline example/webhook.go:59]: minor: I would split the method main in `webhookEnityListExample`, `createWebhookExample` and `webhookListExample`

File: example/webhook.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	for _, webhook := range webhooks.Webhooks {
+		dumpWebhook(webhook, logger)
+	}
+
+}
+func dumpWebhookEntity(webhookEntity model.WebhookEntity, logger *log.Logger) {
+	logger.Println("Name")
+	logger.Printf("SellerSku: %s\n", webhookEntity.Name)
```
Comment (iru-b) [inline example/webhook.go]: please check this `SellerSku` and line above. It is not a sellerSku

File: model/webhook.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	Events []WebhookEntityEvent `json:"Events"`
+}
+
+type WebhookEntityEvent struct {
+	EventName  string `json:"EventName"`
+	EventAlias string `json:"EventAlias"`
+}
+
+func (w *WebhookEntityEvents) UnmarshalJSON(b []byte) error {
+	rawEvents, dataType, _, err := jsonparser.Get(b, "Event")
```
Comment (iru-b) [inline model/webhook.go:58]: I'm not sure. shouldn't it be `jsonparser.Get(b, "Events", "Event")`?

File: model/webhook.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+type WebhookEvents struct {
+	Events []WebhookEvent `json:"Events"`
+}
+
+type WebhookEvent string
+
+func (w *WebhookEvents) UnmarshalJSON(b []byte) error {
+	rawEvents, dataType, _, err := jsonparser.Get(b, "Event")
```
Comment (iru-b) [inline model/webhook.go:137]: I'm not sure. shouldn't it be `jsonparser.Get(b, "Events", "Event")`?

File: resource/webhook.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+type WebhookResource struct {
+	client client.Client
+}
+
+func NewWebhook(client client.Client) WebhookResource {
+	return WebhookResource{client: client}
+}
+
```
Comment (iru-b) [inline resource/webhook.go:17]: please add missing test

**Review by iru-b — CHANGES_REQUESTED** [review]

File: model/webhook.go (hunk 3 above)
Comment (andreasph) [reply model/webhook.go:58]: no, what we have in `b` is `{"Event":...}`

File: model/webhook.go (hunk 4 above)
Comment (andreasph) [reply model/webhook.go:137]: same as above

**Review by iru-b — APPROVED** [review]

**Pull request merged**